topk-js/src/client.rs

```diff
@@ -0,0 +1,227 @@
+use std::sync::Arc;
+
+use napi::bindgen_prelude::*;
+use napi_derive::napi;
+use std::collections::HashMap;
+use topk_protos::v1::control::{self, KeywordIndexType};
+
+#[napi(string_enum)]
+pub enum DataType {
+  Text,
+  Integer,
+  Float,
+  Boolean,
+  F32Vector,
+  U8Vector,
+  BinaryVector,
+  Bytes,
+}
+
+#[napi(string_enum)]
+pub enum VectorFieldIndexMetric {
+  Cosine,
+  Euclidean,
+  DotProduct,
+  Hamming,
+}
+
+impl From<i32> for VectorFieldIndexMetric {
+  fn from(metric: i32) -> Self {
+    match metric {
+      1 => VectorFieldIndexMetric::Cosine,
+      2 => VectorFieldIndexMetric::Euclidean,
+      3 => VectorFieldIndexMetric::DotProduct,
+      4 => VectorFieldIndexMetric::Hamming,
+      _ => VectorFieldIndexMetric::Cosine, // default
+    }
+  }
+}
+
+impl From<VectorFieldIndexMetric> for i32 {
+  fn from(metric: VectorFieldIndexMetric) -> Self {
+    match metric {
+      VectorFieldIndexMetric::Cosine => 1,
+      VectorFieldIndexMetric::Euclidean => 2,
+      VectorFieldIndexMetric::DotProduct => 3,
+      VectorFieldIndexMetric::Hamming => 4,
+    }
+  }
+}
+
+#[napi(string_enum)]
+pub enum EmbeddingDataType {
+  F32,
+  U8,
+  /// Binary quantized uint8
+  Binary,
+}
+
+impl From<Option<i32>> for EmbeddingDataType {
+  fn from(embedding_type: Option<i32>) -> Self {
+    match embedding_type {
+      Some(0) => EmbeddingDataType::F32,
+      Some(1) => EmbeddingDataType::U8,
+      Some(2) => EmbeddingDataType::Binary,
+      // Default to F32 for unspecified embedding type
+      _ => EmbeddingDataType::F32,
+    }
+  }
+}
+
+impl From<EmbeddingDataType> for Option<i32> {
+  fn from(embedding_type: EmbeddingDataType) -> Self {
+    match embedding_type {
+      EmbeddingDataType::F32 => Some(0),
+      EmbeddingDataType::U8 => Some(1),
+      EmbeddingDataType::Binary => Some(2),
+    }
+  }
+}
+
+#[napi]
+pub enum FieldIndex {
+  Keyword,
+  Vector {
+    metric: VectorFieldIndexMetric,
+  },
+  Semantic {
+    model: Option<String>,
+    embedding_type: EmbeddingDataType,
+  },
+}
+
+impl From<control::FieldIndex> for FieldIndex {
+  fn from(field_index: control::FieldIndex) -> Self {
+    match field_index.index.unwrap_or_else(|| {
+      control::field_index::Index::KeywordIndex(control::KeywordIndex { index_type: 0 })
```

Comment:
nit: we shouldn't be using `0` here, but rather an enum. now its hard to understand what option this is, it's also easy to typo this into a different value